### ISSUE TITLE
Bug/Feature: Unable to connect to a Vitess database / Vitess compatibility

### ISSUE THREAD

**Issue by fooforge** [hankinsoft/SQLPro#884]
**Describe the bug**
Vitess is a database clustering system for horizontal scaling of MySQL. There are some [compatibility gaps](https://vitess.io/docs/15.0/reference/compatibility/mysql-compatibility/), but for the most part it's a drop-in replacement. When I try to connect to one of our databases hosted on [PlanetScale](https://planetscale.com/), though, based on SQLPro Studio's log output, the application seems to be able to connect, but the _Connecting_ modal never ends.

**To Reproduce**
1. In your browser, sign up on https://planetscale.com/ and create a free database.
2. On https://app.planetscale.com/, generate new credentials.
3. In SQLPro Studio, create a new connection for your PlanetScale database.
4. In your browser, download the ISRG X1 Root certificate in `.pem` format from https://letsencrypt.org/certs.
5. In SQLPro Studio, click on _Advanced_ -> _SSL Configuration_, select _Enforce_ and add the previously download SSL/TLS root certificate in the _CA certficiate_ field.

**Expected behavior**
I would expect SQLPro Studio to be able to connect to a Vitess database.

**Screenshots**

![CleanShot 2023-02-27 at 10 30 36](https://user-images.githubusercontent.com/959793/221536019-4b9d0190-81f6-449e-bbd9-8ee873bcbf64.png)

**Environment (please complete the following information):**
 - Device: Mac
 - OS: Latest
 - SQLPro Studio app Version: 2023.10 (Build 111008.1)
- Target database server: Vitess

**Additional context**

I see the below in the application's logs which indicates that the SSL handshake has succeeded. No further log output afterwards, however.

```
2023/02/27 09:30:33:915  connectToConnectionId: F718691A-B52B-4CB0-9A28-6FD9C1B0ED94-77448-000001DCCD02D48D
2023/02/27 09:30:33:915  connectToRecentConnectionById (not requesting password)
2023/02/27 09:30:33:915  doConnect: us-east.connect.psdb.cloud (detected port: 3306 specified port: 0)
2023/02/27 09:30:33:929  HSDispatchGroup enter com.hankinsoft.connectionWindowDispatchGroup
2023/02/27 09:30:33:930  doConnect calling getHost:andPort:forServer:userSpecifiedPort:
2023/02/27 09:30:33:930  doConnect self->internalDatabaseConnectionDetails createNewClient
2023/02/27 09:30:33:930  doConnect calling databaseClient->connect
2023/02/27 09:30:33:930  MySQL SSL details are specified.
2023/02/27 09:30:33:939  MySQL going to connect with Authentication: Default, ssl enforced: YES.
2023/02/27 09:30:34:358  MySQL:connect connected with SSL. Cipher is: TLS_AES_128_GCM_SHA256.
```

I have also parsed the query logs on PlanetScale's end to see if SQLPro Studio runs any incompatible queries with Vitess upon a successful connect. It does not seem to get to this stage, however, as I have seen no queries being run by the application at all.

I assume, what happens is that SQLPro Studio verifies the server version first and when it gets back `8.0.23-vitess` it decides to not proceed with the connection.

**NOTE**
If your environmental details above were not completely filled in, the bug report will be closed until they are provided.

/cc https://github.com/planetscale/discussion/discussions/397 (this came in through user feedback)

**Comment by pmsteil** [comment]
I also have this same issue.  I can connect fine from MySQLWorkbench, but not from SQLPro... 

thanks for reporting @fooforge !

**Comment by hankinsoft** [comment]
Reproduced & investigating.

**Comment by hankinsoft** [comment]
Could you try https://d3fwkemdw8spx3.cloudfront.net/mysql/SQLProMySQL.2023.21.app.zip? It's the website version so it will be a trial, but if that sorts it out, I'll get the App Store version of SQLPro Studio submitted to the App Store as well.

**Comment by pmsteil** [comment]
sure

**Comment by pmsteil** [comment]
It worked!!!! I was able to access the PlanetScale database and run a query successfully... :)

**Comment by fooforge** [comment]
@hankinsoft Thanks for jumping on this and implementing support for Vitess. We appreciate it! ❤️

**Comment by pmsteil** [comment]
Yes, thanks @hankinsoft, latest App Store version working for me also :)

**Comment by hankinsoft** [comment]
No worries. FYI if Vitess has any interest in promoting SQLPro, I could offer a discount/promo code or potentially an integration such as a url schema a user could click to add the connection into SQLPro.
